org.eclipse.jdt.core/dom/org/eclipse/jdt/internal/core/dom/ICompilationUnitResolver.java

```diff
@@ -0,0 +1,123 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.jdt.internal.core.dom;
+
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.WorkingCopyOwner;
+import org.eclipse.jdt.core.dom.ASTRequestor;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.FileASTRequestor;
+import org.eclipse.jdt.internal.compiler.batch.FileSystem.Classpath;
+
+/**
+ * This interface is used to resolve a jdt dom tree from source files.
+ * It is contributed to via the compilationUnitResolver extension point.
+ * This interface is currently internal only, and is not considered API.
+ * This interface may be modified, changed, or removed at any time.
+ *
+* <p>
+* <strong>EXPERIMENTAL</strong>. This class or interface has been added as
+* part of a work in progress. There is no guarantee that this API will
+* work or that it will remain the same. Please do not use this API without
+* consulting with the Red Hat team.
+* </p>
+*
+* See https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2641 for discussion on possible
+* changes to this interface
+*
+* @since 3.38
+ */
+public interface ICompilationUnitResolver {
+	/**
+	 * Resolve the given source paths with the following options.
```

Comment:
As I understand it, 'Resolve' includes two steps: parse AST and resolve binding. Can you make the doc to be more specific about what it does?

-> Parse the ASTs and resolve the bindings for the given source files using the following options.

Reply:
should be fixed now.